### DISCUSSION TITLE
How to subclass/derive the DataGrid class

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#14745] [Q&A]
I want to create a subclass of DataGrid, aiming to override OnDataContextEndUpdate.
The non-subclassed DataGrid control is functional.

I created the subclasss MyDataGrid:

```
namespace WinPanel.Views
{
  public partial class MyDataGrid : DataGrid
  {
    public MyDataGrid() : base() { }

    protected override void OnDataContextEndUpdate()
    {
      base.OnDataContextEndUpdate();

      //future: my code here
    }
  }
}
```
And refer to it in the axaml Window section::

`        xmlns:v="using:WinPanel.Views"`

and in the containing panel :

          <v:MyDataGrid>
           .
           .
          </v:MyDataGrid

This compiles, but the (My)DataGrid shows no data.

What else is needed to make the derived class do what the DataGrid does as a starting point?

**Comment by stevemonaco** [comment]
`DataGrid` is a `TemplatedControl`. As such, it gets its appearance from a `ControlTheme`. You need to override `StyleKeyOverride` so that your derived control will use the default `DataGrid` `ControlTheme`. See [this doc](https://docs.avaloniaui.net/docs/basics/user-interface/styling/styles#style-key) for an example.

**Comment by wgraaf** [comment]
Thanks! 
Adding 

 `   protected override Type StyleKeyOverride => typeof(DataGrid);`

to the derived class made it work.
